### PR TITLE
test(api): add unit tests for Pi harness adapter config generation

### LINKED ISSUE #1362: Add unit tests for `services/harness/pi.py` config generator
## Background

The Pi config generator (`observal-server/services/harness/pi.py`) produces MCP configs, skill files, and hook scripts for Pi users, but has no dedicated unit tests.

## What needs to be done

- Create `tests/test_harness_pi_config_generator.py`.
- Mock the database layer and test:
  - MCP config generation produces valid JSON matching Pi expected schema.
  - Skill file generation writes correctly formatted SKILL.md.
  - Hook script generation includes expected shebang and event wiring.
  - Edge case: empty component list produces no files (no crash).

## Files to touch

- `tests/test_harness_pi_config_generator.py` *(new)*

## Reference

- `observal-server/services/harness/pi.py`
- Existing patterns in `tests/test_agent_config_generator.py`

## Acceptance criteria

- [ ] At least 4 test cases.
- [ ] `pytest tests/test_harness_pi_config_generator.py -v` passes.
- [ ] No Docker required (all externals mocked).
- [ ] Ruff lint passes.

### PR DESCRIPTION
**Pull request by kilqwe**
## Purpose / Description
`PiAdapter` in `observal-server/services/harness/pi.py` had no dedicated unit tests. This PR adds coverage for MCP config generation, skill file path rewriting, agent profile generation, and edge cases.

## Fixes
* Fixes #1362 

## Approach
Created `tests/test_harness_pi_config_generator.py` with 17 unit tests covering:

- MCP config generation produces valid JSON with correct mcpServers schema
- Skill file paths include both agent name and skill name after path rewriting
- Agent profile (AGENTS.md) generation writes correct path and content
- Edge cases: empty/None inputs produce empty dict with no crash, no files written

All external dependencies mocked via MagicMock. No Docker required.

## How Has This Been Tested?
uv run pytest tests/test_harness_pi_config_generator.py -v
# 17 passed in 0.40s

uvx ruff check tests/test_harness_pi_config_generator.py
# All checks passed

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)

## AI Assistance
_Was generative AI tooling used to co-author this PR?_

- [x] Yes(Please Specify the tool): Claude Sonnet 4.6 (Low)
- [x] Was the generated code manually reviewed and tested?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 <!-- SPDX-FileCopyrightText: 2026 Shaan Narendran <shaannaren06@gmail.com> -->
 <!-- SPDX-FileCopyrightText: 2026 SrihariLegend <sriharilegend23@gmail.com> -->
 <!-- SPDX-FileCopyrightText: 2026 DoomsCoder <vedantkakade05@gmail.com> -->
+<!-- SPDX-FileCopyrightText: 2026 kilqwe <shreyas0514@gmail.com> -->
```
Comment (Haz3-jolt) [inline CHANGELOG.md]: Don't edit the changelog

File: docs/cli/prompt.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 <!-- SPDX-FileCopyrightText: 2026 Gokulkrishnan <gokulkri247@gmail.com> -->
+<!-- SPDX-FileCopyrightText: 2026 kilqwe <shreyas0514@gmail.com> -->
 <!-- SPDX-License-Identifier: AGPL-3.0-only -->
```
Comment (Haz3-jolt) [inline docs/cli/prompt.md]: Unnecessary touch

File: docs/cli/registry.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <!-- SPDX-FileCopyrightText: 2026 Apoorv Garg <apoorvgarg.21@gmail.com> -->
 <!-- SPDX-FileCopyrightText: 2026 Hari Srinivasan <harisrini21@gmail.com> -->
```
Comment (Haz3-jolt) [inline docs/cli/registry.md:2]: Same here

File: docs/cli/skill.md [hunk 4]
```diff
@@ -1,4 +1,5 @@
 <!-- SPDX-FileCopyrightText: 2026 Nithin-Bhargav-07 <gaddamnithinbhargav@gmail.com> -->
+<!-- SPDX-FileCopyrightText: 2026 kilqwe <shreyas0514@gmail.com> -->
 <!-- SPDX-License-Identifier: AGPL-3.0-only -->
```
Comment (Haz3-jolt) [inline docs/cli/skill.md]: And here

File: infra/terraform/aws/variables.tf [hunk 5]
```diff
@@ -478,4 +479,3 @@ variable "alb_scheme" {
     error_message = "alb_scheme must be 'internet-facing' or 'internal'."
   }
 }
```
Comment (Haz3-jolt) [inline infra/terraform/aws/variables.tf:480]: .

File: observal_cli/skills/observal/SKILL.md [hunk 6]
```diff
@@ -47,7 +48,7 @@ observal agent pull AGENT_NAME --harness kiro --no-prompt --dir .
 ```
 
 **For Pi (`--harness pi`):**
-When pulling for Pi, the CLI natively downloads the agent's files into isolated profiles at `~/.pi/agent/agents/<AGENT_NAME>`. 
+When pulling for Pi, the CLI natively downloads the agent's files into isolated profiles at `~/.pi/agent/agents/<AGENT_NAME>`.
```
Comment (Haz3-jolt) [inline observal_cli/skills/observal/SKILL.md]: It's the same why?

File: packages/pi-extension/extensions/observal.ts [hunk 7]
```diff
@@ -198,7 +199,7 @@ export default function (pi: ExtensionAPI) {
 
       try {
         applyProfile(choice);
-        
+
         if (state?.config) {
```
Comment (Haz3-jolt) [inline packages/pi-extension/extensions/observal.ts:202]: Unnecessary again

**Review by Haz3-jolt — CHANGES_REQUESTED** [review]
Please remove all changes from our of scope files

**Comment by kilqwe** [comment]
Hi, Sorry for the changes in the previous commit, The pre-commit hook auto-fixed all trailing spaces and EOF issues and I must have accidentally staged all changes during commit. I have now pushed a clean commit which touches only the newly created `tests/test_harness_pi_config_generator.py`. All the other changes have been restored to original state.

**Review by Haz3-jolt — APPROVED** [review]

**Comment by kilqwe** [comment]
Thank you for the review and merge!
